### PR TITLE
Support non-standard mongo port

### PR DESCRIPTION
**Pull request by steven-schattenberg-itential**

### CODE REVIEW

**Review by Madias2222 — APPROVED** [review]

File: roles/gateway/tasks/main.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2024, Itential, Inc
+Z# Copyright (c) 2024, Itential, Inc
```
Comment (kvelarde-itential) [inline roles/gateway/tasks/main.yml:1]: I
```suggestion
# Copyright (c) 2024, Itential, Inc
```

Comment (kvelarde-itential) [reply roles/gateway/tasks/main.yml]: It looks like you got a random Z char here...

File: roles/mongodb/tasks/configure-selinux.yml [hunk 2]
```diff
@@ -8,6 +8,16 @@
...
+    - name: Allow mongodb to listen on tcp port when using non-standard mongo port
+      when:
+        - ansible_selinux.status == "enabled"
+        - mongo_port != 27017
+      community.general.seport:
+        ports: "{{ mongo_port }}"
+        proto: tcp
+        setype: mongod_port_t
+        state: present
+
```
Comment (kvelarde-itential) [inline roles/mongodb/tasks/configure-selinux.yml:20]:
```suggestion
    - name: Allow mongodb to listen on tcp port when using non-standard mongo port
      community.general.seport:
        ports: "{{ mongo_port }}"
        proto: tcp
        setype: mongod_port_t
        state: present
      when:
        - ansible_selinux.status == "enabled"
        - mongo_port != 27017
```

Comment (kvelarde-itential) [reply roles/mongodb/tasks/configure-selinux.yml:20]: The when clause should only be first when using a block.

**Review by Nick-Andreano — DISMISSED** [review]

File: roles/mongodb/tasks/main.yml [hunk 3]
```diff
@@ -200,6 +200,7 @@
     # This creates the admin user that has root access to the database
     - name: Add admin user to database
       community.mongodb.mongodb_user:
+        login_port: "{{ mongo_port }}"
         login_user: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary('admin', omit) }}"
         login_password: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary(mongo_user_admin_password, omit) }}"
```
Comment (kvelarde-itential) [inline roles/mongodb/tasks/main.yml:205]:
```suggestion
        login_user: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary('admin', omit) }}"
        login_password: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary(mongo_user_admin_password, omit) }}"
        login_port: "{{ mongo_port }}"        
```

File: roles/mongodb/tasks/main.yml [hunk 4]
```diff
@@ -218,6 +219,7 @@
     # itential database.  It is used by IAP to connect to the db.
     - name: Add itential user to database
       community.mongodb.mongodb_user:
+        login_port: "{{ mongo_port }}"
         login_user: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary('admin', omit) }}"
         login_password: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary(mongo_user_admin_password, omit) }}"
```
Comment (kvelarde-itential) [inline roles/mongodb/tasks/main.yml:224]:
```suggestion
        login_user: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary('admin', omit) }}"
        login_password: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary(mongo_user_admin_password, omit) }}"
        login_port: "{{ mongo_port }}"
```

File: roles/mongodb/tasks/main.yml [hunk 5]
```diff
@@ -236,6 +238,7 @@
     # LocalAAA database.  It is used by the local AAA adapter to login to IAP.
     - name: Add localaaa user to database
       community.mongodb.mongodb_user:
+        login_port: "{{ mongo_port }}"
         login_user: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary('admin', omit) }}"
         login_password: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary(mongo_user_admin_password, omit) }}"
```
Comment (kvelarde-itential) [inline roles/mongodb/tasks/main.yml:243]:
```suggestion
        login_user: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary('admin', omit) }}"
        login_password: "{{ mongodb_auth_enabled is defined and mongodb_auth_enabled | ternary(mongo_user_admin_password, omit) }}"
        login_port: "{{ mongo_port }}"
```

**Review by kvelarde-itential — CHANGES_REQUESTED** [review]

**Review by kvelarde-itential — APPROVED** [review]

**Pull request merged**